spec/jobs/threat_metrix_js_verification_job_spec.rb

```diff
@@ -0,0 +1,193 @@
+require 'rails_helper'
+
+RSpec.describe ThreatMetrixJsVerificationJob, type: :job do
+  let(:proofing_device_profiling_collecting_enabled) { true }
+  let(:threatmetrix_org_id) { 'ABCD1234' }
+  let(:threatmetrix_session_id) { 'some-session-id' }
+
+  let(:threatmetrix_signing_key) do
+    OpenSSL::PKey::RSA.new 2048
+  end
+
+  let(:threatmetrix_signing_cert_expiry) { Time.zone.now + 3600 }
+
+  let(:threatmetrix_signing_certificate) do
+    if threatmetrix_signing_key.present?
```

Comment:
minor nit: personally I would avoid too much conditional logic inside these `let` helpers, and just override them as needed in particular specs